mkosi/__init__.py

```diff
@@ -3466,12 +3456,19 @@ def invoke_repart(
 
     output = json.loads(run(cmdline, stdout=subprocess.PIPE).stdout)
 
+    roothash = usrhash = None
     for p in output:
-        if p["type"].startswith("usr") or p["type"].startswith("root"):
-            usr_only = p["type"].startswith("usr")
-            return (p["label"], p.get("roothash"), usr_only)
+        if (h := p.get("roothash")) is None:
+            continue
+
+        assert p["type"].startswith("usr") or p["type"].startswith("root")
```

Comment:
We use `asserts` in many places, but I'd like to get rid of them. They can be compiled out and their error is a bit nondescript. 
```suggestion
        if no (p["type"].startswith("usr") or p["type"].startswith("root")):
            raise MkosiException(f"Unknown verity partition type {p['type']}")
```